src/hooks/wallets/mpc/__tests__/useMPC.test.ts

```diff
@@ -0,0 +1,218 @@
+      state: {
+        get: () => ({
+          wallets: [],
+          walletModules: [],
+        }),
+      },
+    } as any)
+    jest.spyOn(useChains, 'useCurrentChain').mockReturnValue({
+      chainId: '5',
+      chainName: 'Goerli',
+      blockExplorerUriTemplate: {
+        address: 'https://goerli.someprovider.io/{address}',
+        txHash: 'https://goerli.someprovider.io/{txHash}',
+        api: 'https://goerli.someprovider.io/',
+      },
+      nativeCurrency: {
+        decimals: 18,
+        logoUri: 'https://logo.goerli.com',
+        name: 'Goerli ETH',
+        symbol: 'ETH',
+      },
+      rpcUri: {
+        authentication: RPC_AUTHENTICATION.NO_AUTHENTICATION,
+        value: 'https://goerli.somerpc.io',
+      },
+    } as any)
+
+    const mockWeb3AuthMpcCoreKit = jest.spyOn(require('@web3auth/mpc-core-kit'), 'Web3AuthMPCCoreKit')
+    const mockProvider = jest.fn()
+    mockWeb3AuthMpcCoreKit.mockImplementation(() => {
+      return new MockMPCCoreKit(require('@web3auth/mpc-core-kit').COREKIT_STATUS.INITIALIZED, mockProvider)
+    })
+
+    renderHook(() => useInitMPC())
+
+    await waitFor(() => {
+      expect(connectWalletSpy).toBeCalled()
+      expect(getMPCCoreKitInstance()).toBeDefined()
+    })
+  })
+
+  it('should copy event handlers and emit chainChanged if the current chain is updated', async () => {
+    const connectWalletSpy = jest.fn().mockImplementation(() => Promise.resolve())
+    jest.spyOn(useOnboard, 'connectWallet').mockImplementation(connectWalletSpy)
+    jest.spyOn(useOnboard, 'getConnectedWallet').mockReturnValue({
+      address: hexZeroPad('0x1', 20),
+      label: ONBOARD_MPC_MODULE_LABEL,
+      chainId: '1',
+      provider: {} as any,
+    })
+    jest.spyOn(useOnboard, 'default').mockReturnValue({
+      state: {
+        get: () => ({
+          wallets: [],
+          walletModules: [],
+        }),
+      },
+    } as any)
+    jest.spyOn(useChains, 'useCurrentChain').mockReturnValue({
+      chainId: '5',
+      chainName: 'Goerli',
+      blockExplorerUriTemplate: {
+        address: 'https://goerli.someprovider.io/{address}',
+        txHash: 'https://goerli.someprovider.io/{txHash}',
+        api: 'https://goerli.someprovider.io/',
+      },
+      nativeCurrency: {
+        decimals: 18,
+        logoUri: 'https://logo.goerli.com',
+        name: 'Goerli ETH',
+        symbol: 'ETH',
+      },
+      rpcUri: {
+        authentication: RPC_AUTHENTICATION.NO_AUTHENTICATION,
+        value: 'https://goerli.somerpc.io',
+      },
+    } as any)
+
+    const mockWeb3AuthMpcCoreKit = jest.spyOn(require('@web3auth/mpc-core-kit'), 'Web3AuthMPCCoreKit')
+    const mockChainChangedListener = jest.fn()
+    const mockProviderBefore = {
+      listeners: (eventName: string) => {
+        if (eventName === 'chainChanged') {
+          return [mockChainChangedListener]
+        }
+      },
+    }
+
+    setMPCCoreKitInstance({
+      provider: mockProviderBefore,
+    } as any)
```

Comment:
Can we get rid of this `any`?

Reply:
I can only switch it for a `as unknown as Web3AuthMPCCoreKit` which is the same in this context.
I don't want to mock the entire object as it's quite complex and we only use this one field of it in the hook.

Reply:
I'd suggest opting for that over `any`.